packages/apps/tests/test_http_stream.py

```diff
@@ -63,37 +61,45 @@ def http_stream(self, mock_api_client, conversation_reference, mock_logger):
         return HttpStream(mock_api_client, conversation_reference, mock_logger)
 
     @pytest.mark.asyncio
-    async def test_stream_emit_message_flushes_after_500ms(self, mock_api_client, conversation_reference, mock_logger):
-        """Test that messages are flushed after 500ms timeout."""
+    async def test_stream_emit_message_flushes_immediately(self, mock_api_client, conversation_reference, mock_logger):
+        """Test that messages are flushed immediately."""
 
         stream = HttpStream(mock_api_client, conversation_reference, mock_logger)
-        start_time = datetime.now()
-
         stream.emit("Test message")
-        await asyncio.sleep(0.6)  # Wait longer than 500ms timeout
-
-        assert mock_api_client.send_call_count > 0, "Should have sent at least one message"
-        assert any(t >= start_time + timedelta(milliseconds=450) for t in mock_api_client.send_times), (
-            "Should have waited approximately 500ms before sending"
-        )
+        await asyncio.sleep(0.07)  # Wait for the flush task to complete
+        assert mock_api_client.send_call_count == 1
 
     @pytest.mark.asyncio
-    async def test_stream_multiple_emits_restarts_timer(self, mock_api_client, conversation_reference, mock_logger):
+    async def test_stream_multiple_emits_timer_check(self, mock_api_client, conversation_reference, mock_logger):
         """Test that multiple emits reset the timer."""
 
         stream = HttpStream(mock_api_client, conversation_reference, mock_logger)
 
         stream.emit("First message")
-        await asyncio.sleep(0.3)  # Wait less than 500ms
-
-        stream.emit("Second message")  # This should reset the timer
-        await asyncio.sleep(0.3)  # Still less than 500ms from second emit
-        assert mock_api_client.send_call_count == 0, "Should not have sent yet"
-        await asyncio.sleep(0.3)  # Now over 500ms from second emit
-        assert mock_api_client.send_call_count > 0, "Should have sent messages after timer expired"
+        stream.emit("Second message")
+        stream.emit("Third message")
+        stream.emit("Fourth message")
+        stream.emit("Fifth message")
+        stream.emit("Sixth message")
+        stream.emit("Seventh message")
+        stream.emit("Eighth message")
+        stream.emit("Ninth message")
+        stream.emit("Tenth message")
+        stream.emit("Eleventh message")
+        stream.emit("Twelfth message")
+
+        await asyncio.sleep(0.07)  # Wait for the flush task to complete
```

Comment:
Instead of adding all these magic numbers, can we simply mock `call_later`? 

Basically you want to simulate how call_later will behave right? so mock that function, and have it trigger on some signal.

emit
emit
signal() <- this is your test to simulate the passing of some time. Not real time, but simulated time.
verify behavior 

Essentially, you know time is going to progress. You're not testing the behavior of call_later. So this is a reasonable way to test stuff like this.